float32 max_distance
float32 step
uint16[] pixel_distances

